src/sentry/static/sentry/app/components/accordion.jsx

```diff
@@ -0,0 +1,84 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import styled from 'react-emotion';
+import InlineSvg from 'app/components/inlineSvg';
+import space from 'app/styles/space';
+import {Flex} from 'grid-emotion';
+import Badge from 'app/components/badge';
+import Button from 'app/components/buttons/button';
+
+class AccordionButton extends React.Component {
+  static propTypes = {
+    labelClosed: PropTypes.string.isRequired,
+    labelOpen: PropTypes.string,
+    cutoff: PropTypes.number,
+    children: PropTypes.arrayOf(PropTypes.node).isRequired,
+  };
+
+  static defaultProps = {
+    cutoff: 0,
+  };
+
+  constructor(props, context) {
+    super(props, context);
+
+    this.state = {
+      isOpen: false,
+    };
+  }
+
+  handleClick = e => {
+    e.preventDefault();
+    this.setState({isOpen: !this.state.isOpen});
+  };
+
+  render() {
+    const {labelOpen, labelClosed, cutoff} = this.props;
+    const children = React.Children.toArray(this.props.children);
+
+    if (children.length <= cutoff + 1) return children;
+
+    return (
+      <React.Fragment>
+        {children.slice(0, cutoff)}
+        <StyledButton onClick={this.handleClick} open={this.state.isOpen}>
+          <Flex align="center" justify="space-between" w={1}>
+            <Flex align="center">
+              {this.state.isOpen ? labelOpen || labelClosed : labelClosed}
+              <Badge style={{marginLeft: space(0.25)}} text={children.length - cutoff} />
```

Comment:
Was hoping styled components would cut down on inline styles :'(

Reply:
Wait why can't this just be a `StyledBadge` ?

Reply:
I kind of like inline styles for one offs (it's like bem but without the overhead) but i'm not gonna argue w more styled components so lez do it